assignments/closure.js

```diff
@@ -1,21 +1,47 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
+let myName = "Emily";
+let yourName = "Bob";
+greeter = () => console.log(`Hello ${yourName}! My name is ${myName}.`);
 
```

Comment:
This isn't strictly a closure.

Although the greeter function has closure over the myName and yourName variables, the point of creating a closure is to use a function (the outer function) that returns another function (the inner function). 

The function that is returned (the inner function) maintains the lexical environment of the outer function. 

A good example of the usefulness of this is creating factory functions, for example:
```
function adder(x) {
    function sum(y) {
        return x + y
    }

   return sum;
}
```

These are useful because you can use them to create multiple other functions. For example if i do:

```
const add3 = adder(3);
```
I then have a function called add3 that will always add 3 to whatever number I call it with.

Tbh, it's a fairly confusing concept - something that I think many experienced JS developers don't understand - and the fact that we covered it on day one of JS is fairly amusing. I hope this make sense, probably not the best explanation...!